setup.py

```diff
@@ -1,12 +1,46 @@
 """Setuptools package definition."""
 
+import distutils.cmd
+import os
+
 from setuptools import find_packages, setup
 
 version = {}
 with open("caluma/caluma_metadata.py") as fp:
     exec(fp.read(), version)
 
+pipenv_setup = """
+echo UID=$(id --user) > .env
+echo ENV=dev >> .env
+docker-compose up -d db
+rm -f Pipfile*
+touch Pipfile
+pipenv install --python 3.6 -r requirements.txt
+pipenv install -d -r requirements-dev.txt
```

Comment:
Doing this will will flag all the deps in `requirements.txt` as dev-dependencies as well, because we have the line `-r requirements.txt` in there.
Why would you want to remove the Pipfile and Pipefile.lock during the setup?
If you want to check if the Pipfile.lock is up-to-date, there is the `--deploy` flag (or `--sync`?) that checks that, no?
https://pipenv.readthedocs.io/en/latest/advanced/#using-pipenv-for-deployments

Reply:
> Doing this will will flag all the deps in `requirements.txt` as dev-dependencies as well, because we have the line `-r requirements.txt` in there.

I am aware of how to use pipenv correctly, we just can't do it (yet?). For new project I use pipenv in its full glory.

It does't matter that the non-dev dependencies are added to the dev-dependencies, since you can't install only the dev-dependencies with pipenv anyways. This is the best way to bootstrap the Pipfile from requirements*.txt.

If you do

```bash
pipenv --rm
pipenv sync
```
you get only the non-dev-requirements.

```bash
pipenv --rm
pipenv sync -d
```
you get the dev- and the non-dev-requirement.

> Why would you want to remove the Pipfile and Pipfile.lock during the setup?

Well, I don't want to get rid of the requirements*.txt file (yet?), a lot of tooling expects the requirements*.txt.

* Because we still have the requirements.txt as primary definition, we can't use pipenv for deployments
* Because we use docker for deployment, we don't want to use pipenv for deployments

> If you want to check if the Pipfile.lock is up-to-date, there is the `--deploy` flag (or `--sync`?) that checks that, no?
> https://pipenv.readthedocs.io/en/latest/advanced/#using-pipenv-for-deployments

Well the idea is to call ./setup.py pipenv only once for your local repository. From then on you use only pipenv commands. Only if the requirements*.txt have changed you would call ./setup.py pipenv again. Since the Pipfile is meant to be the primary source, it doesn't update constaints from requirements.txt if Pipfile exists, so we have to delete the Pipfile*, in order to transfer the changes to the Pipfile*.

Reply:
Ah I see, makes sense now. :)
Just saw that Pipfile* is in the .gitignore, so it's only used as an alternative workflow.
I like!